frontend/js/src/mbid-mapping/MBIDMappingModal.tsx

```diff
@@ -250,9 +266,11 @@ export default NiceModal.create(({ listenToMap }: MBIDMappingModalProps) => {
               ) : (
                 <div className="card listen-card">
                   <SearchTrackOrMBID
+                    key={defaultValue}
```

Comment:
Hm, I think maybe I spoke too soon :)

After trying it, the issue is that if I click the button (text is copied as expected), then clear the search input text, then click the button again, nothing happens because the defaultValue hasn't changed in the parent so the `key` here remains the same.

If the value changed in the parent, it would re-render the SearchTrackOrMBID, but as it stands it won't work as we want.
I'm still not comfortable with the useEffect solution you had to react to changes in defaultValue, because it seems like the kind of mechanism that could overwrite the user's input text without a clear action from the user.

One suggestion I can think of si that we can make the `key` change on each button click.
We can add a new state for a number value defaulting to 0, and in `copyTextToSearchField` we increment the value.
Then the `SearchTrackOrMBID` key would be composed from these two bits of data:
```key={`${defaultValue}-${copyTextClickCounter}`}```

What do you think? It seems a bit convoluted compared to the useEffect you used previously, but it ensures we don't update the input unless there is a clear user action…

Reply:
Ya, I think this method is also good. I have implemented it and committed the changes.